### PR TITLE
rabbitmq_user: Append user permissions

### LINKED ISSUE #19659: rabbitmq_user should not update permissions each time
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
rabbitmq_user

##### ANSIBLE VERSION
```
ansible 2.2.0.0
```

##### SUMMARY
If you are going to update permissions with items in the vhost options, all the permissions will be overwritten

##### STEPS TO REPRODUCE
use the playbook below

```yaml
- rabbitmq_user:
    user: "{{ rabbit_user }}"
    password: "{{ rabbit_password }}"
    permissions:
      - vhost: "{{ item }}"
        configure_priv: .*
        read_priv: .*
        write_priv: .*
  with_items:
    - "{{ vhosts }}"
```

##### EXPECTED RESULTS
a user should has permissions for all {{ vhosts }}

##### ACTUAL RESULTS
a user has permissions for the last {{ vhosts }} value

### PR DESCRIPTION
**Pull request by jackivanov**
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
rabbitmq_user

##### ANSIBLE VERSION
```
ansible 2.2.0.0
```

##### SUMMARY
Added a new comparison, if `state=append`, then don't need to clear existing permissions.

### CODE REVIEW

File: lib/ansible/modules/messaging/rabbitmq_user.py [hunk 1]
```diff
@@ -102,10 +102,10 @@
     choices: [ "yes", "no" ]
   state:
     description:
-      - Specify if user is to be added or removed
+      - Specify if user is to be added, removed or append permissions
```
Comment (gundalow) [inline lib/ansible/modules/messaging/rabbitmq_user.py]: `Specify if user is to be added, removed or append permission. Note I(append) was added in Ansible 2.3.`

**Comment by gundalow** [comment]
needs_revision

**Comment by jctanner** [comment]
@gunph1ld are you able to implement the requested changes so that we can get this merged?

needs_info

**Comment by jackivanov** [comment]
@jctanner Yes, just pushed a few changes

**Comment by jackivanov** [comment]
Hi Folks,

Would this be merged? Anything else needs from me?

ping @jctanner @gundalow

**Pull request closed without merging**